core/src/main/java/io/questdb/cairo/TableWriter.java

```diff
@@ -1562,12 +1724,135 @@ public void warmUp() {
+            }
+
+            if (metadata.getId() != detachedMetadata.getId()) {
+                // very same table, no foreign partition attaching is allowed
+                throw CairoException.detachedMetadataMismatch("table_id");
+            }
+            if (metadata.getTimestampIndex() != detachedMetadata.getTimestampIndex()) {
+                // designated timestamps in both tables, same index
+                throw CairoException.detachedMetadataMismatch("timestamp_index");
+            }
+
+            // check column name, type and isIndexed
+            detachedDeleteExtraColNames.clear(); // .attachable may have extra columns we do not need
+            detachedAddMissingColNames.clear();  // .attachable may lack columns that we need to add
+            boolean colsNegativeTypeMatch = false;
+            for (int i = 0; i < columnCount; i++) {
+                int colIdx = metadata.getWriterIndex(i);
+                String columnName = metadata.getColumnName(colIdx);
+
+                // check name
+                int detColIdx = detachedMetadata.getColumnIndexQuiet(columnName);
+                if (detColIdx == -1) {
+                    throw CairoException.detachedMetadataMismatch("missing_column=" + columnName);
+                }
+
+                // check type
+                int colType = metadata.getColumnType(colIdx);
+                int detColType = detachedMetadata.getColumnType(detColIdx);
+                if (colType != detColType) {
+                    if (colType == -detColType) { // a column was deleted
+                        if (colType < detColType) {
+                            detachedDeleteExtraColNames.add(columnName);
+                        } else {
+                            // column was added to the table, and does not exist in attaching
+                            detachedAddMissingColNames.add(columnName);
+                        }
+                        colsNegativeTypeMatch = true;
+                    } else {
+                        throw CairoException.detachedColumnMetadataMismatch(colIdx, columnName, "type");
+                    }
+                }
+
+                // check is indexed
+                if (metadata.isColumnIndexed(colIdx) != detachedMetadata.isColumnIndexed(detColIdx)) {
+                    throw CairoException.detachedColumnMetadataMismatch(colIdx, columnName, "is_indexed");
+                }
+            }
+
+            // check structure version
+            if (colsNegativeTypeMatch) {
+                if (metadata.getStructureVersion() == detachedMetadata.getStructureVersion()) {
+                    throw CairoException.detachedMetadataMismatch("structure_version should be different");
+                }
+            } else if (metadata.getStructureVersion() != detachedMetadata.getStructureVersion()) {
+                throw CairoException.detachedMetadataMismatch("structure_version");
```

Comment:
Any column add, delete will bump structure version, it's very surprising tests work with this strict check.